stake-pool/js/src/index.ts

```diff
@@ -0,0 +1,168 @@
+import borsh from 'borsh';
+import solanaWeb3 from '@solana/web3.js';
+
+export class StakePool {
+  /**
+   * Wrapper class for a stake pool.
+   * Each stake pool has a stake pool account and a validator list account.
+   * (Optionally) a stake pool can also have a name and ticker.
+   */
+  name: string;
+  ticker: string;
+  stakePool: decodedStakePool;
+  validatorList: decodedValidatorList;
+}
+
+export interface decodedStakePool {
+  pubkey: solanaWeb3.PublicKey;
+  account: solanaWeb3.AccountInfo<schema.StakePoolAccount>;
+}
+
+export interface decodedValidatorList {
+  pubkey: solanaWeb3.PublicKey;
+  account: solanaWeb3.AccountInfo<schema.ValidatorListAccount>;
+}
+
+async function getStakePoolAccount(
+  connection: solanaWeb3.Connection,
+  stakePoolPubKey: solanaWeb3.PublicKey,
+): Promise<decodedStakePool> {
+  /**
+   * Retrieves and deserializes a StakePool account using a web3js connection and the stake pool address.
+   * @param connection: An active web3js connection.
+   * @param stakePoolPubKey: The public key (address) of the stake pool account.
+   */
+  try {
+    const account = await connection.getAccountInfo(stakePoolPubKey);
+
+    return {
+      pubkey: stakePoolPubKey,
+      account: {
+        data: schema.StakePoolAccount.decode(account.data),
+        executable: account.executable,
+        lamports: account.lamports,
+        owner: account.owner,
+      },
+    };
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+async function getValidatorListAccount(
+  connection: solanaWeb3.Connection,
+  validatorListPubKey: solanaWeb3.PublicKey,
+): Promise<decodedValidatorList> {
+  /**
+   * Retrieves and deserializes a ValidatorList account using a web3js connection and the validator list address.
+   * @param connection: An active web3js connection.
+   * @param validatorListPubKey: The public key (address) of the validator list account.
+   */
+  try {
+    const account = await connection.getAccountInfo(validatorListPubKey);
+
+    return {
+      pubkey: validatorListPubKey,
+      account: {
+        data: schema.ValidatorListAccount.decode(account.data),
+        executable: account.executable,
+        lamports: account.lamports,
+        owner: account.owner,
+      },
+    };
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+async function getStakePoolAccounts(
+  connection: solanaWeb3.Connection,
+  stakePoolProgramAddress: solanaWeb3.PublicKey,
+): Promise<(decodedStakePool | decodedValidatorList)[]> {
+  /**
+   * Retrieves all StakePool and ValidatorList accounts that are running a particular StakePool program.
+   * @param connection: An active web3js connection.
+   * @param stakePoolProgramAddress: The public key (address) of the StakePool program.
+   */
+  try {
+    let response = await connection.getProgramAccounts(stakePoolProgramAddress);
+
+    const stakePoolAccounts = response.map(a => {
+      let decodedData;
+
+      if (a.account.data.readUInt8() === 1) {
```

Comment:
For a next phase (v2?), we can do this even better :smile: we should probably have all of our state structs given as:
```rs
pub struct State {
    Uninitialized,
    StakePool(StakePool),
    ValidatorList(ValidatorList),
}
```
And then we can just call deserialize and we're good